### DISCUSSION TITLE
Pimcore 10 method overriding for classification store(classificationsearch) is having an issue with route

### DISCUSSION THREAD

**Discussion by manojthotla25** [pimcore/pimcore#10742] [Q&A / Support]
Hi Team,

Task : I am trying to add columns in the collections search. I have created a bundle and inside startup.js i have used the below code. and in the below code inside collection search if it is this.config.fieldname == "listOfProductAttribute" then i am using route 'pimcore_admin_dataobject_classificationstore_collectionsactionget2'; the same route i have defined in my controller also.
But i after installing the bundle i am getting route doesnot exist. Please check below for my controller and routing file.

Is there some change with route in pimcore 10 version because my colleague says it was working for him in pimcore 6.8

pimcore.registerNS("pimcore.plugin.ClassificationStoreBundle");

pimcore.plugin.ClassificationStoreBundle = Class.create(pimcore.plugin.admin, {
    getClassName: function () {
        return "pimcore.plugin.ClassificationStoreBundle";
    },

    initialize: function () {
        pimcore.plugin.broker.registerPlugin(this);
    },

    pimcoreReady: function (params, broker) {
        // alert("ClassificationStoreBundle ready!");

        Ext.override(pimcore.object.classificationstore.keySelectionWindow, {
            getGridPanel: function () {

                var postFix;
                var route;
                var nameWidth = 200;
                var descWidth = 590;

                if (this.config.isCollectionSearch) {
                    if (this.config.fieldname == "listOfProductAttribute") {
                        // alert("test");
                        route = 'pimcore_admin_dataobject_classificationstore_collectionsactionget2';
                        this.groupFields = ['id', 'name', 'description', 'coll_name'];
                    }
                    else {
                        route = 'pimcore_admin_dataobject_classificationstore_collectionsactionget';
                        this.groupFields = ['id', 'name', 'description'];
                    }
                }
                else if (this.config.isGroupSearch) {
                    route = 'pimcore_admin_dataobject_classificationstore_groupsactionget';
                    this.groupFields = ['id', 'name', 'description'];
                } else if (this.config.isGroupsBySearch) {
                    this.groupFields = ['id', 'groupName', 'keyName', 'keyDescription', 'keyId', 'groupId'];
                } else {
                    route = 'pimcore_admin_dataobject_classificationstore_propertiesget';
                    nameWidth = 150;
                    descWidth = 490;
                    this.groupFields = ['id', 'groupName', 'name', 'description'];
                }

                if (this.config.isGroupByKeySearch) {
                    var url = Routing.generate('pimcore_admin_dataobject_classificationstore_searchrelations');
                } else {
                    var url = Routing.generate(route);
                }

                var readerFields = [];
                for (var i = 0; i < this.groupFields.length; i++) {
                    readerFields.push({ name: this.groupFields[i] });
                }

                var gridColumns = [];
                if (this.config.isGroupByKeySearch) {
                    gridColumns.push({ text: "ID", width: 60, sortable: true, dataIndex: 'id' });

                    gridColumns.push({
                        text: t("group"),
                        flex: 1,
                        sortable: true,
                        dataIndex: 'groupName',
                        filter: 'string',
                        renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                    });

                    gridColumns.push({
                        text: t("name"),
                        flex: 1,
                        sortable: true,
                        dataIndex: 'keyName',
                        filter: 'string',
                        renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                    });

                    gridColumns.push({
                        text: t("description"),
                        flex: 1,
                        sortable: true,
                        dataIndex: 'keyDescription',
                        filter: 'string',
                        renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                    });
                } else {
                    gridColumns.push({ text: "ID", width: 40, sortable: true, dataIndex: 'id' });

                    if (postFix == "properties") {
                        gridColumns.push({
                            text: t("classificationstore_tag_col_group"),
                            width: 150,
                            sortable: true,
                            dataIndex: 'groupName'
                        });
                    }

                    gridColumns.push({
                        text: t("name"),
                        width: nameWidth,
                        sortable: true,
                        dataIndex: 'name',
                        renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                    });

                    if (this.config.fieldname == "listOfProductAttribute") {
                        gridColumns.push({
                            text: t("Class Name"),
                            width: nameWidth,
                            sortable: true,
                            dataIndex: 'coll_name',
                            renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                        });
                    }

                    gridColumns.push({
                        text: t("description"),
                        width: descWidth,
                        sortable: true,
                        dataIndex: 'description',
                        renderer: pimcore.helpers.grid.getTranslationColumnRenderer.bind(this)
                    });
                }

                var extraParams = {
                    storeId: this.config.storeId
                };

                if (this.config.frameName) {
                    extraParams.frameName = this.config.frameName;
                }

                var proxy = {
                    type: 'ajax',
                    url: url,
                    reader: {
                        type: 'json',
                        rootProperty: 'data'
                    },
                    extraParams: extraParams
                };

                if (this.config.object) {
                    proxy.extraParams.oid = this.config.object.id;
                    proxy.extraParams.fieldname = this.config.fieldname;
                }

                this.store = new Ext.data.Store({
                    remoteSort: true,
                    proxy: proxy,
                    fields: readerFields
                });

                var pageSize = pimcore.helpers.grid.getDefaultPageSize(-1);
                this.pagingtoolbar = pimcore.helpers.grid.buildDefaultPagingToolbar(this.store, { pageSize: pageSize });

                this.gridPanel = new Ext.grid.Panel({
                    store: this.store,
                    border: false,
                    columns: gridColumns,
                    loadMask: true,
                    columnLines: true,
                    bodyCls: "pimcore_editable_grid",
                    stripeRows: true,
                    selModel: Ext.create('Ext.selection.RowModel', {
                        mode: 'MULTI'
                    }),
                    bbar: this.pagingtoolbar,
                    listeners: {
                        rowdblclick: function (grid, record, tr, rowIndex, e, eOpts) {
                            if (this.config.isGroupByKeySearch) {
                                let data = [grid.getStore().getAt(rowIndex).get("groupId")];
                                this.addGroups(data);
                            } else {
                                let data = [grid.getStore().getAt(rowIndex).id];

                                if (this.config.isCollectionSearch) {
                                    this.addCollections(data);
                                } else if (this.config.isGroupSearch) {
                                    this.addGroups(data);
                                } else {
                                    this.addKeys(data);
                                }
                            }

                            this.searchWindow.close();
                        }.bind(this)
                    }
                });

                this.store.load();

                this.resultPanel.removeAll();
                this.resultPanel.add(this.gridPanel);
                this.resultPanel.updateLayout();
            },

        })
    }
});

var ClassificationStoreBundlePlugin = new pimcore.plugin.ClassificationStoreBundle();

********************************************************************************************************************

Controller:
class ClassficationOverrideController extends ClassificationstoreController
{

   /**
     * @Route("/collections", name="pimcore_admin_dataobject_classificationstore_collectionsactionget2", methods={"GET"})
     *
     * @param Request $request
     *
     * @return JsonResponse
     */
    public function collectionsActionGet(Request $request)
    {
        $this->checkPermission('objects');

        $start = 0;
        $limit = $request->get('limit') ? $request->get('limit') : 15;

        $orderKey = 'name';
        $order = 'ASC';

        if ($request->get('dir')) {
            $order = $request->get('dir');
        }

        if ($request->get('limit')) {
            $limit = $request->get('limit');
        }
        if ($request->get('start')) {
            $start = $request->get('start');
        }

        $allParams = array_merge($request->request->all(), $request->query->all());
        $sortingSettings = \Pimcore\Bundle\AdminBundle\Helper\QueryParams::extractSortingSettings($allParams);
        if ($sortingSettings['orderKey'] && $sortingSettings['order']) {
            $orderKey = $sortingSettings['orderKey'];
            $order = $sortingSettings['order'];
        }

        if ($request->get('overrideSort') == 'true') {
            $orderKey = 'id';
            $order = 'DESC';
        }

        $storeIdFromDefinition = 0;
        $allowedCollectionIds = [];
        if ($request->get('oid')) {
            $object = DataObject\Concrete::getById($request->get('oid'));
            $class = $object->getClass();
            /** @var DataObject\ClassDefinition\Data\Classificationstore $fd */
            $fd = $class->getFieldDefinition($request->get('fieldname'));
            $allowedGroupIds = $fd->getAllowedGroupIds();

            if ($allowedGroupIds) {
                $db = \Pimcore\Db::get();
                $query = 'select * from classificationstore_collectionrelations where groupId in (' . implode(',', $allowedGroupIds) .')';
                $relationList = $db->fetchAll($query);

                if (is_array($relationList)) {
                    foreach ($relationList as $item) {
                        $allowedCollectionIds[] = $item['colId'];
                    }
                }
            }

            $storeIdFromDefinition = $fd->getStoreId();
        }

        $query = "SELECT count(cls_s_grp.id) as cls_count, cls_s_grp.id, cls_s_grp.name as coll_name, cls_s_grp.storeId, cls_s_grp.description, cls_s_col.name  
        FROM `classificationstore_groups` as cls_s_grp 
        join classificationstore_collectionrelations as cls_s_rel on (cls_s_grp.id = cls_s_rel.groupId)
        join classificationstore_collections as cls_s_col on (cls_s_col.id = cls_s_rel.colId)";
        $conditionParts = [];
        $db = Db::get();

        $searchfilter = $request->get('searchfilter');
        if ($searchfilter) {
            $conditionParts[] = '(cls_s_col.name LIKE ' . $db->quote('%' . $searchfilter . '%') . ' OR cls_s_grp.description LIKE ' . $db->quote('%'. $searchfilter . '%') . ' or cls_s_grp.name LIKE ' . $db->quote('%' . $searchfilter . '%') . ' )';

        }

        $storeId = $request->get('storeId');
        $storeId = $storeId ? $storeId : $storeIdFromDefinition;

        $conditionParts[] = '(cls_s_col.storeId = ' . $storeId . ')';

        if ($request->get('filter')) {
            $filterString = $request->get('filter');
            $filters = json_decode($filterString);

            foreach ($filters as $f) {
                $conditionParts[] = "cls_s_col.".$db->quoteIdentifier($f->property) . ' LIKE ' . $db->quote('%' . $f->value . '%');
            }
        }

        if ($allowedCollectionIds) {
            $conditionParts[] = ' cls_s_col.id in (' . implode(',', $allowedCollectionIds) . ')';
        }
        
        $condition = implode(' AND ', $conditionParts);
        $condition = " Where ".$condition;
        $query .= $condition;

        $query .= " group by cls_s_grp.name";

        $query .= " Order by $orderKey $order";

        $queryMain = $query;
        $query .= " limit $start,$limit";

        $configList = $db->fetchAll($query);
        $list = $db->fetchAll($queryMain);
        
        $rootElement = [];

        $data = [];
      
        foreach ($configList as $config) {
            $name = $config["name"];
            if (!$name) {
                $name = 'EMPTY';
            }
            $item = [
                'storeId' => $config["storeId"],
                'id' => $config["id"],
                'name' => $name,
                'description' => $config["description"],
                'coll_name' => $config["coll_name"],
            ];
            if ($config["creationDate"]) {
                $item['creationDate'] = $config["creationDate"];
            }

            if ($config["modificationDate"]) {
                $item['modificationDate'] = $config["modificationDate"];
            }

            $data[] = $item;
        }
        $rootElement['data'] = $data;
        $rootElement['success'] = true;
        $rootElement['total'] = count($list);
        return $this->adminJson($rootElement);
    }

***************************************************************************************************************************

routing.yml:

pimcore_admin_dataobject_classificationstore_collectionsactionget2:
    resource: "@ClassificationStoreBundle/Controller/"
    type: annotation
    prefix: /admin

**Comment by dvesh3** [comment]
@manojthotla25 Hi, please try adding `option.expose` to your route config:

```
pimcore_admin_dataobject_classificationstore_collectionsactionget2:
resource: "@ClassificationStoreBundle/Controller/"
type: annotation
prefix: /admin
options:
        expose: true
```
